lib/src/browser_adapter.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library w_transport.src.browser_adapter;
+
+import 'dart:async';
+
+import 'package:w_transport/src/http/browser/client.dart';
+import 'package:w_transport/src/http/browser/requests.dart';
+import 'package:w_transport/src/http/client.dart';
+import 'package:w_transport/src/http/requests.dart';
+import 'package:w_transport/src/platform_adapter.dart';
+import 'package:w_transport/src/web_socket/client/w_socket.dart';
+import 'package:w_transport/src/web_socket/w_socket.dart';
+
+/// Adapter for the browser platform. Exposes factories for all of the transport
+/// classes that return browser-specific implementations that leverage
+/// dart:html.
+class BrowserAdapter implements PlatformAdapter {
+  /// Construct a new [BrowserClient] instance that implements [Client].
+  Client newClient() => new BrowserClient();
+
+  /// Construct a new [BrowserFormRequest] instance that implements
+  /// [FormRequest].
+  FormRequest newFormRequest() => new BrowserFormRequest();
+
+  /// Construct a new [BrowserJsonRequest] instance that implements
+  /// [JsonRequest].
+  JsonRequest newJsonRequest() => new BrowserJsonRequest();
+
+  /// Construct a new [BrowserMultipartRequest] instance that implements
+  /// [MultipartRequest].
+  MultipartRequest newMultipartRequest() => new BrowserMultipartRequest();
+
+  /// Construct a new [BrowserPlainTextRequest] instance that implements
+  /// [Request].
+  Request newRequest() => new BrowserPlainTextRequest();
+
+  /// Construct a new [BrowserStreamedRequest] instance that implements
+  /// [StreamedRequest].
+  StreamedRequest newStreamedRequest() => new BrowserStreamedRequest();
+
+  /// Construct a new [ClientWSocket] instance that implements [WSocket].
+  Future<WSocket> newWSocket(Uri uri,
+          {Iterable<String> protocols, Map<String, dynamic> headers}) =>
+      ClientWSocket.connect(uri, protocols: protocols, headers: headers);
```

Comment:
if you're choosing to rename `client` to `browser` in most places, should `ClientWSocket` also change?